### ISSUE TITLE
Regression in v4.0.0 in Spirit of Speed 1937

### ISSUE THREAD

**Issue by xesarni** [samuelgr/Xidi#24]
Xidi v.4.0.0 does work only partially in Spirit of Speed 1937. It is possible to remap buttons in the game's settings but during racing the controller doesn't work at all. Xidi v3.1.0 works fine.

Windows 10 21H1 x64, 8Bitdo SN30 Pro+

**Comment by samuelgr** [comment]
Could you please provide a debug-level log with both Xidi 3.1.0 and Xidi 4.0.0?

**Comment by xesarni** [comment]
[3.1 - Xidi_DInput_sos.exe_5532.log](https://github.com/samuelgr/Xidi/files/8884729/3.1.-.Xidi_DInput_sos.exe_5532.log)
[4.0 - Xidi_DInput_sos.exe_8784.log](https://github.com/samuelgr/Xidi/files/8884730/4.0.-.Xidi_DInput_sos.exe_8784.log)

**Comment by samuelgr** [comment]
Xidi v3.1 doesn't support force feedback, so the game has no trouble acquiring the virtual controller device because Xidi treats all acquisitions as no-ops. Every time it tries, the result is 0x00000000 (which maps to `DI_OK`).

```
[06/11/2022 22:48:04] [I] Invoked Xidi::VirtualDirectInputDevice<0>::Acquire() on Xidi virtual controller 1, result = 0x00000000.
```

Xidi v4.0 introduces a separation between exclusive mode and nonexclusive mode, which is required by force feedback. Having looked at the logs, I suspect the game is creating multiple controller objects and trying to acquire all of them in exclusive mode. Maybe DirectInput allows this (and "exclusive mode" means "exclusive to the application" rather than "exclusive to the specific object") but Xidi currently does not.

From the logs:

```
[06/11/2022 22:48:49] [I] Acquiring Xidi virtual controller 1 in exclusive mode.
[06/11/2022 22:48:49] [I] Invoked Xidi::VirtualDirectInputDevice<0>::Acquire() on Xidi virtual controller 1, result = 0x00000000.
...
[06/11/2022 22:48:55] [I] Acquiring Xidi virtual controller 1 in exclusive mode.
[06/11/2022 22:48:55] [W] Invoked Xidi::VirtualDirectInputDevice<0>::Acquire() on Xidi virtual controller 1, result = 0x80070005.
...
[06/11/2022 22:48:59] [I] Acquiring Xidi virtual controller 1 in exclusive mode.
[06/11/2022 22:48:59] [W] Invoked Xidi::VirtualDirectInputDevice<0>::Acquire() on Xidi virtual controller 1, result = 0x80070005.
```

The first attempt succeeds, but it seems like the game does not use the first object and tries instead to use others. The second and third attempts fail with code 0x80070005 (which maps to `DIERR_OTHERAPPHASPRIO` or alternatively `E_ACCESSDENIED`). Assuming this is the root cause, the fix is to eliminate exclusivity between controller objects.

**Comment by samuelgr** [comment]
This issue should be fixed by 2ca5716c6c7ee1bc29cf31861d4199b150f74142. Please try the attached development build and let me know if it works. If not, please supply a debug log. Unfortunately I do not have access to this game so I cannot test it myself.

[Xidi-v4.0.2-dev.8.0+2ca5716c.zip](https://github.com/samuelgr/Xidi/files/9174522/Xidi-v4.0.2-dev.8.0%2B2ca5716c.zip)

**Comment by xesarni** [comment]
It works. Thanks!
